src/Orleans/Async/TaskExtensions.cs

```diff
@@ -130,11 +131,13 @@ internal static T WaitForResultWithThrow<T>(this Task<T> task, TimeSpan timeout)
                 return;
             }
 
-            await Task.WhenAny(taskToComplete, Task.Delay(timeout));
+            var timeoutCancellationTokenSource = new CancellationTokenSource();
```

Comment:
Should I fill new issue and make another PR? [Asynchronously wait for Task<T> to complete with timeout](http://stackoverflow.com/questions/4238345/asynchronously-wait-for-taskt-to-complete-with-timeout#comment27036701_11191070)